frontend/src/app/main/data/workspace/common.cljs

```diff
@@ -280,40 +281,23 @@
               {:keys [frame-id parent-id]} shape]
           [frame-id parent-id (inc index)])))))
 
-(defn add-shape-changes
-  ([page-id objects selected attrs]
-   (add-shape-changes page-id objects selected attrs true))
-  ([page-id objects selected attrs reg-object?]
-   (let [id    (:id attrs)
-         shape (gpr/setup-proportions attrs)
-
-         default-attrs (if (= :frame (:type shape))
-                         cp/default-frame-attrs
-                         cp/default-shape-attrs)
-
-         shape    (merge default-attrs shape)
-
-         not-frame? #(not (= :frame (get-in objects [% :type])))
-         selected (into #{} (filter not-frame?) selected)
-
-         [frame-id parent-id index] (get-shape-layer-position objects selected attrs)
-
-         redo-changes  (cond-> [{:type :add-obj
-                                 :id id
-                                 :page-id page-id
-                                 :frame-id frame-id
-                                 :parent-id parent-id
-                                 :index index
-                                 :obj shape}]
-                         reg-object?
-                         (conj {:type :reg-objects
-                                :page-id page-id
-                                :shapes [id]}))
-         undo-changes  [{:type :del-obj
-                         :page-id page-id
-                         :id id}]]
-
-     [redo-changes undo-changes])))
+(defn make-new-shape
+  [attrs objects selected]
+  (let [default-attrs (if (= :frame (:type attrs))
+                        cp/default-frame-attrs
+                        cp/default-shape-attrs)
+
+        selected-non-frames
+        (into #{} (filter #(not= (:type (get objects %)) :frame) selected))
```

Comment:
probably this can be better and will be more readable:

```clojure
(into #{} (comp (map (d/getf objects))
                (remove cph/frame-shape?))
      selected)
```

Reply:
okkk